ui/app/src/components/Preview/Preview.tsx

```diff
@@ -139,7 +139,7 @@ function Preview() {
       // Not the first render. Something changed and we're updating.
 
       // Check if either the QS or the state has changed
-      const qsSiteChanged = qs.site !== prev.qsSite || qs.site !== site;
+      const qsSiteChanged = qs.site !== prev.qsSite && qs.site !== site;
```

Comment:
This goes against what the comment is saying. Need you to walk through the logic of why this is the correct behaviour and how it doesn't break something else.